app/models/setting/email.rb

```diff
@@ -0,0 +1,74 @@
+require 'facter'
+
+class Setting::Email < Setting
+  NON_EMAIL_YAML_SETTINGS = %w(send_welcome_email email_reply_address email_subject_prefix)
+
+  def self.load_defaults
+    Foreman::Deprecation.deprecation_warning('1.15', "'email.yaml' file configuration has been deprecated, email configuration has been moved to Foreman settings") if mailconfig.present?
+    # Check the table exists
+    return unless super
+    domain = Facter.value(:domain) || SETTINGS[:domain]
+    email_reply_address = "Foreman-noreply@#{domain}"
+
+    self.transaction do
+      [
+        self.set('email_reply_address', N_("Email reply address for emails that Foreman is sending"), email_reply_address, N_('Email reply address')),
+        self.set('email_subject_prefix', N_("Prefix to add to all outgoing email"), '[foreman]', N_('Email subject prefix')),
+        self.set('send_welcome_email', N_("Send a welcome email including username and URL to new users"), false, N_('Send welcome email')),
+        self.set('delivery_method', N_("Method used to deliver email"), 'sendmail', N_('Delivery method'), nil, { :collection => Proc.new {{:sendmail => N_("Sendmail"), :smtp => N_("SMTP")}}}),
+        self.set('smtp_enable_starttls_auto', N_("SMTP automatic STARTTLS"), false, N_('SMTP enable StartTLS auto')),
+        self.set('smtp_tls', N_("Enable SSL/TLS"), false, N_('SMTP use SSL/TLS')),
+        self.set('smtp_address', N_("Address to connect to"), '', N_('SMTP address')),
+        self.set('smtp_port', N_("Port to connect to"), 25, N_('SMTP port')),
+        self.set('smtp_domain', N_("HELO/EHLO domain"), '', N_('SMTP HELO/EHLO domain')),
+        self.set('smtp_user_name', N_("Username to use to authenticate, if required"), '', N_('SMTP username')),
+        self.set('smtp_password', N_("Password to use to authenticate, if required"), '', N_('SMTP password'), nil, {:encrypted => true}),
+        self.set('smtp_authentication', N_("Specify authentication type, if required"), 'none', N_('SMTP authentication'), nil, { :collection => Proc.new {{:plain => 'plain', :login => 'login', :cram_md5 => 'cram_md5', '' => N_("none")}}}),
+        self.set('sendmail_arguments', N_("Specify additional options to sendmail"), '-i', N_('Sendmail arguments')),
+        self.set('sendmail_location', N_("The location of the sendmail executable"), "/usr/sbin/sendmail", N_('Sendmail location'))
+      ].each { |s| self.create! s.update(:category => "Setting::Email")}
+    end
+    true
+  end
+
+  def self.config_file
+    File.expand_path(File.join(File.dirname(__FILE__), '..', '..', '..', 'config', 'email.yaml'))
+  end
+
+  def self.delivery_settings
+    options = {}
+    self.all.each do |setting|
+      if Setting[:delivery_method].to_s == "smtp" && setting.name.start_with?('smtp') && setting.value.present?
```

Comment:
Using `setting.value.present?` is omitting settings that are `false` - e.g. set enable StartTLS auto to false and it isn't returned in the `Setting::Email.delivery_settings` return value so the setting has no effect. (Because it defaults to `true` in Rails.)